src/main/scala/nl.knaw.dans.easy.deposit/Submitter.scala

```diff
@@ -104,13 +108,19 @@ class Submitter(stagingBaseDir: File,
       _ = stageBag.addMetadataFile(msg4DataManager, s"$depositorInfoDirectoryName/message-from-depositor.txt")
       _ <- stageBag.addMetadataFile(agreementsXml, s"$depositorInfoDirectoryName/agreements.xml")
       _ <- stageBag.addMetadataFile(datasetXml, "dataset.xml")
-      _ <- stageBag.addMetadataFile(filesXml, "files.xml")
-      _ <- workerActions(draftDeposit.id, draftBag, stageBag, submitDir)
+      _ <- stageBag.addMetadataFile(filesXml, "files.xml") // TODO stress test with large number of files?
+      agreementData = AgreementData(user, datasetMetadata)
+      agreement <- agreementGenerator.generate(agreementData, draftDeposit.id)
+      attachments = Map("agreement.pdf" -> Mailer.pdfDataSource(agreement),
+        "metadata.xml" -> Mailer.xmlDataSource(datasetXml), // TODO XMLs serialized here as well as by stageBag.addMetadataFile
+        "files.xml" -> Mailer.xmlDataSource(filesXml))
```

Comment:
Not sure we should send `files.xml`, it might be too big to mail, or even `dataset.xml`. It was just easy to to as-is.